CMakeLists.txt

```diff
@@ -46,9 +46,9 @@ else() # GCC/clang/mingw
 #  -fwrapv              -> Make signed integer overflows defined
 #  -fvisibility=hidden	-> Force defaultsymbol visibility to hidden
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -fno-strict-aliasing -fwrapv -fvisibility=hidden")
-
 # Use -O2 as maximum optimization level. -O3 has it's problems with yquake2.
-string(REPLACE "-O3" "-O2" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+set(CMAKE_BUILD_TYPE RelWithDebInfo)
```

Comment:
While RelWithDebInfo defaults to `-O2` on most platforms it's not guaranteed. It isn't even guaranteed that RelWithDebInfo has lower optimization level than Release. Therefore we can't rely on RelWithDebInfo not setting -O3.